### PR TITLE
V2: fix and add tests for (multitask) classification 

### PR DESCRIPTION
**Pull request by hwpang**
## Description
This PR fixes classification and adds relevent tests.

## Example / Current workflow
Currently, training a classification model fails at `_default_metric` not defined as it is currently commented out. After uncommenting `AUROCMetric`, it fails with `task` position variable is required.

## Bugfix / Desired workflow
- Added an integration test for classification. The test currently fails with a `nan` loss. Still trying to figure it out.
- Added CLI test for training and predicting classification.
- Added a `task` flag for `AUROCMetric` as it's a required input.

## Checklist
- [ ] linted with flake8?
- [ ] (if appropriate) unit tests added?

### CODE REVIEW

File: tests/integration/test_classification_mol.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        not_nan = ~targets.isnan()
+        predss.append(preds)
+        targetss.append(targets)
+        masks.append(not_nan)
+
+    preds = torch.cat(predss)
+    targets = torch.cat(targetss)
+    mask = torch.cat(masks)
+    auroc = F.auroc(preds[mask], targets[mask].long(), task="binary")
+    assert auroc >= 0.95
```
Comment (KnathanM) [inline tests/integration/test_classification_mol.py:76]: Why 0.95? Why not 0.99?

File: chemprop/nn/metrics.py [hunk 2]
```diff
@@ -122,8 +122,14 @@ def __call__(self, preds: Tensor, targets: Tensor, mask: Tensor, *args, **kwargs
...
 
+    def __init__(self, task: str) -> None:
+        self.task = task
+
     def __call__(self, preds: Tensor, targets: Tensor, mask: Tensor, *args, **kwargs):
-        return F.auroc(preds[mask], targets[mask].long())
+        return F.auroc(preds[mask], targets[mask].long(), task=self.task)
+    
+    def forward(self, preds, targets, mask) -> Tensor:
+        return self.__call__(preds, targets, mask)
```
Comment (KnathanM) [inline chemprop/nn/metrics.py:132]: Do we have an idea where in the code this `forward()` method would be called? I have a feeling that because it's signature doesn't match the abstract one defined in `class Metric` then if part of the code does try to call this forward, there will be an error. Maybe we just don't implement a forward method.

**Review by KnathanM — APPROVED** [review]
